### DISCUSSION TITLE
Sidebar filter

### DISCUSSION THREAD

**Discussion by geerew** [withastro/starlight#973] [Feature Requests]
### What version of `starlight` are you using?

0.8.0

### What is your idea?

Some side menus can become quite large. It would be neat to have a filter input that as the user types, filters out non-relevant menu items.

![Screenshot 2023-08-29 at 11 58 12](https://github.com/withastro/starlight/assets/494875/f181804f-23e4-4827-95aa-38b39c38ae53)

### Why is this feature necessary?

Some side menus can become quite large

### Do you have examples of this feature in other projects?

- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_certificate_authority_certificate

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by fwx5618177** [comment]
I think has added this filter function.

**Comment by delucis** [comment]
Thanks for the suggestion @geerew! This would probably need really thorough testing and has a lot of UX and accessibility considerations, so definitely not a simple thing to implement.

I think I might push for us to wait until #415 is addressed and then it might be something people can implement as a plugin rather than having it as a built-in feature.

**Comment by HiDeoo** [comment]
To give an update on that feature request:

- [Component overrides](https://starlight.astro.build/guides/overriding-components/) are now available in Starlight for a while
- Overrides can [re-use](https://starlight.astro.build/guides/overriding-components/#reuse-a-built-in-component) a built-in component
- [`<Sidebar>`](https://starlight.astro.build/reference/overrides/#sidebar-1) is one of the overridable components
- The sidebar content is accessible in overrides using the [`Astro.props.sidebar`](https://starlight.astro.build/reference/overrides/#sidebar) props

Considering such filtering can now be entirely implemented by users in their Starlight project (and has been done by some users already) and that there is no plan to have it be a built-in feature, I'll close this discussion.
